### PR TITLE
tree-wide: remove unused type argument in TAKE_GENERIC() macro

### PR DESCRIPTION
**Pull request by Integral-Tech**
The `type` argument in `TAKE_GENERIC()` macro is redundant.

### CODE REVIEW

File: src/fundamental/macro-fundamental.h [hunk 1]
```diff
@@ -376,18 +376,16 @@
...
-                type _var_ = *_pvar_;                            \
-                type _nullvalue_ = nullvalue;                    \
-                *_pvar_ = _nullvalue_;                           \
-                _var_;                                           \
+#define TAKE_GENERIC(var, nullvalue)                 \
+        ({                                           \
+                typeof(var) *_pvar_ = &(var);        \
+                typeof(var) _var_ = *_pvar_;         \
+                typeof(var) _nullvalue_ = nullvalue; \
+                *_pvar_ = _nullvalue_;               \
```
Comment (YHNdnzj) [inline src/fundamental/macro-fundamental.h:384]: The extra `_nullvalue_` var is not really useful here. Just assign to pvar directly if you touch this?

**Comment by Integral-Tech** [comment]
@YHNdnzj Fixed :)

File: src/fundamental/macro-fundamental.h (hunk 1 above)
Comment (yuwata) [reply src/fundamental/macro-fundamental.h:384]: That's for type safety, I think.

Comment (keszybz) [reply src/fundamental/macro-fundamental.h:384]: Yeah. With automatic typing we'd accept almost anything. So we don't want to do this.

**Comment by Integral-Tech** [comment]
@keszybz But in systemd codebase, `type` argument is actually not used.

File: src/fundamental/macro-fundamental.h [hunk 2]
```diff
@@ -376,18 +376,16 @@
 
 /* Takes inspiration from Rust's Option::take() method: reads and returns a pointer, but at the same time
  * resets it to NULL. See: https://doc.rust-lang.org/std/option/enum.Option.html#method.take */
-#define TAKE_GENERIC(var, type, nullvalue)                       \
-        ({                                                       \
-                type *_pvar_ = &(var);                           \
```
Comment (keszybz) [inline src/fundamental/macro-fundamental.h:381]: It's used here.

Comment (Integral-Tech) [reply src/fundamental/macro-fundamental.h:381]: > It's used here.

In my PR, I just changed `TAKE_GENERIC` macro to `#define TAKE_GENERIC(var, nullvalue)`, and `type` is replaced with `typeof(var)`.

Comment (keszybz) [reply src/fundamental/macro-fundamental.h:381]: Yes, and this removes the type check. Before, we explicitly specified the type. So for example:
```C
const char *x = "asdf";
TAKE_GENERIC(y, int, NULL);
```
does NOT compile. With your patch there is no type check, and `TAKE_GENERIC(y, NULL)` is fine.

Comment (Integral-Tech) [reply src/fundamental/macro-fundamental.h:381]: Okay, I see

**Pull request closed without merging**